components/public-api-server/pkg/apiv1/idp.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package apiv1
+
+import (
+	"context"
+	"fmt"
+
+	connect "github.com/bufbuild/connect-go"
+	v1 "github.com/gitpod-io/gitpod/components/public-api/go/experimental/v1"
+	"github.com/gitpod-io/gitpod/components/public-api/go/experimental/v1/v1connect"
+	"github.com/gitpod-io/gitpod/public-api-server/pkg/proxy"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus/ctxlogrus"
+	"github.com/zitadel/oidc/pkg/oidc"
+)
+
+type IDTokenSource interface {
+	IDToken(ctx context.Context, org string, audience []string, userInfo oidc.UserInfo) (string, error)
+}
+
+func NewIDPService(serverConnPool proxy.ServerConnectionPool, source IDTokenSource) *IDPService {
+	return &IDPService{
+		connectionPool: serverConnPool,
+		idTokenSource:  source,
+	}
+}
+
+type IDPService struct {
+	connectionPool proxy.ServerConnectionPool
+	idTokenSource  IDTokenSource
+
+	v1connect.UnimplementedWorkspacesServiceHandler
+}
+
+var _ v1connect.IDPServiceHandler = ((*IDPService)(nil))
+
+// GetIDToken implements v1connect.IDPServiceHandler
+func (srv *IDPService) GetIDToken(ctx context.Context, req *connect.Request[v1.GetIDTokenRequest]) (*connect.Response[v1.GetIDTokenResponse], error) {
+	workspaceID, err := validateWorkspaceID(req.Msg.GetWorkspaceId())
+	if err != nil {
+		return nil, err
+	}
+
+	if len(req.Msg.Audience) < 1 {
+		return nil, connect.NewError(connect.CodeInvalidArgument, fmt.Errorf("must have at least one audience entry"))
```

Comment:
```suggestion
		return nil, connect.NewError(connect.CodeInvalidArgument, fmt.Errorf("Must have at least one audience entry"))
```
For consistency with other APIs here